common/src/lib.rs

```diff
@@ -170,7 +170,14 @@ impl Game {
     }
 
     fn cast_vote(&mut self, player_id: UserId, vote: Vote) {
-        if let Some(ref mut story) = self.selected_story {
+        if matches!(
+            &self.selected_story,
+            Some(story) if !story.votes_revealed
+        ) {
+            let story = self
+                .selected_story
+                .as_mut()
+                .unwrap(/* safe because of check above */);
             story.add_vote(player_id, vote);
         }
```

Comment:
:art: the following looks a bit simpler and also avoids the unwrapping
```suggestion
        if let Some(ref mut story) = self.selected_story {
            if !story.votes_revealed {
                story.add_vote(player_id, vote);
            }
        }
```

Reply:
true